augur/curate/format_dates.py

```diff
@@ -0,0 +1,237 @@
+#!/usr/bin/env python3
+"""
+Format date fields to ISO 8601 dates (YYYY-MM-DD), where incomplete dates
+are masked with 'XX' (e.g. 2023 -> 2023-XX-XX).
+"""
+import re
+from datetime import datetime
+
+from augur.errors import AugurError
+from augur.io.print import print_err
+from augur.types import DataErrorMethod
+
+
+def register_parser(parent_subparsers):
+    parser = parent_subparsers.add_parser("format-dates",
+        parents=[parent_subparsers.shared_parser],
+        help=__doc__)
+
+    required = parser.add_argument_group(title="REQUIRED")
+    required.add_argument("--date-fields", nargs="+",
+        help="List of date field names in the record that need to be standardized.")
+    required.add_argument("--expected-date-formats", nargs="+",
+        help="Expected date formats that are currently in the provided date fields, " +
+             "defined by standard format codes as listed at " +
+             "https://docs.python.org/3/library/datetime.html#strftime-and-strptime-format-codes. " +
+             "If a date string matches multiple formats, it will be parsed as the first format in the list.")
+
+    optional = parser.add_argument_group(title="OPTIONAL")
+    optional.add_argument("--failure-reporting",
+        type=DataErrorMethod,
+        choices=[ method for method in DataErrorMethod ],
+        default=DataErrorMethod.ERROR_FIRST,
+        help="How should failed date formatting be reported.")
+    optional.add_argument("--no-mask-failure", dest="mask_failure",
+        action="store_false",
+        help="Do not mask dates with 'XXXX-XX-XX' and return original date string if date formatting failed.")
```

Comment:
Seems like the auto-generated docs for `store_false` actions can be misleading:

> --no-mask-failure
>    Do not mask dates with ‘XXXX-XX-XX’ and return original date string if date formatting failed.
>    Default: True

Reply:
You can do:

```
help="Do not mask dates… Default: False%(default).0s"
```

to work around this with the standard argparse help formatter.

See [the explanation of why this works](https://github.com/nextstrain/cli/blob/017c53805e8317951327d24c04184615cc400b09/nextstrain/cli/argparse.py#L13-L21).

Reply:
Thanks for suggesting the workaround! Added in https://github.com/nextstrain/augur/pull/1146/commits/c0e3084a3581f4bcec17631fd96e6c4f0ef06464